### PR TITLE
Batchsize + large data + scan issues

### PR DESCRIPTION
**Pull request by javmorcas**

### CODE REVIEW

File: srcCpp/RTIDDSImpl.cxx [hunk 1]
```diff
@@ -679,54 +689,58 @@ bool RTIDDSImpl<T>::ParseConfig(int argc, char *argv[])
...
+     * so in that case, we just disabled batching, else, the customer set it up,
+     * so we explitly fail
      */
-    } else if (_BatchSize > 0 && (unsigned long)_BatchSize < _DataLen * 2) {
-            /*
-             * We don't want to use batching if the batch size is not large
-             * enough to contain at least two samples (in this case we avoid the
-             * checking at the middleware level).
-             */
+    if (_isLargeData && _BatchSize > 0) {
```
Comment (softwarejimenez) [inline srcCpp/RTIDDSImpl.cxx]:
```
if (_BatchSize > 0) {
        if (_LatencyTest) {
            if (isBatchSizeProvided) {
                fprintf(stderr, "Batching cannot be used with Latency test.\n");
                return false;
            } else {
                _BatchSize = 0; //Disable Batching
            }
        }
        if (_IsAsynchronous) {
            fprintf(stderr, "Batching cannot be used with asynchronous writing.\n");
            return false;
        }
        /*
         * Large Data + batching cannot be set. But batching is enabled by default,
         * so in that case, we just disabled batching, else, the customer set it up,
         * so we explitly fail
         */
        if (_isLargeData) {
            if (isBatchSizeProvided) {
                fprintf(stderr, "Batching cannot be used with asynchronous writing.\n");
                return false;
            } else {
                _BatchSize = -2;
            }
        }
        /*
         * We don't want to use batching if the batch size is not large
         * enough to contain at least two samples (in this case we avoid the
         * checking at the middleware level).
         */
        else if ((unsigned long) _BatchSize < _DataLen * 2) {
            if (isBatchSizeProvided || _isScan) {
                /*
                 * Batchsize disabled. A message will be print if _batchsize < 0 in
                 * perftest_cpp::PrintConfiguration()
                 */
                _BatchSize = -1;
            }
            else {
                _BatchSize = 0;
            }
        }
    }
```

**Comment by softwarejimenez** [comment]
Hi @javmorcas , Can you take a look into the feedback?

File: srcCpp/RTIDDSImpl.cxx [hunk 2]
```diff
@@ -679,54 +689,58 @@ bool RTIDDSImpl<T>::ParseConfig(int argc, char *argv[])
...
-        if (_scan_max_size > (unsigned long)(std::min)(MAX_SYNCHRONOUS_SIZE,MAX_BOUNDED_SEQ_SIZE)) {
-            if (_useUnbounded == 0) {
-                _useUnbounded = MAX_BOUNDED_SEQ_SIZE;
-            }
-            _isLargeData = true;
-        } else {
-            _useUnbounded = 0;
-            _isLargeData = false;
+    /* Check if we need to enable Large Data. This works also for -scan */
+    if (_DataLen > (unsigned long)(std::min)(
```
[the remark this answers is not in the document]
Comment (javmorcas) [reply srcCpp/RTIDDSImpl.cxx]: Fixed, but `(std::min)(` <- together, as this is a function, not a cast

File: srcCpp/RTIDDSImpl.cxx [hunk 3]
```diff
@@ -679,54 +689,58 @@ bool RTIDDSImpl<T>::ParseConfig(int argc, char *argv[])
...
 
-    if (_isLargeData && _BatchSize > 0) {
-            fprintf(stderr, "Batching cannot be used with asynchronous writing.\n");
-            return false;
+        /*
+         * We don't want to use batching if the batch size is not large
+         * enough to contain at least two samples (in this case we avoid the
+         * checking at the middleware level).
+         */
+        if ((unsigned long)_BatchSize < _DataLen * 2) {
```
[the remark this answers is not in the document]
Comment (javmorcas) [reply srcCpp/RTIDDSImpl.cxx]: Done.

File: srcJava/com/rti/perftest/ddsimpl/RTIDDSImpl.java [hunk 4]
```diff
@@ -1634,67 +1645,78 @@ else if ("-noPositiveAcks".toLowerCase().startsWith(argv[i].toLowerCase())) {
...
-             * checking at the middleware level).
+             * Large Data + batching cannot be set. But batching is enabled by default,
+             * so in that case, we just disabled batching, else, the customer set it up,
+             * so we explitly fail
              */
-            if (isBatchSizeProvided) {
+            if (_isLargeData) {
+                if (isBatchSizeProvided) {
+                    System.err.println(
+                        "Batching cannot be used with asynchronous writing.");
```
Comment (softwarejimenez) [inline srcJava/com/rti/perftest/ddsimpl/RTIDDSImpl.java]: add 4 spaces

File: srcJava/com/rti/perftest/harness/PerfTest.java [hunk 5]
```diff
@@ -837,10 +837,16 @@ private void printConfiguration() {
                 sb.append(" Bytes (Use \"-batchSize 0\" to disable batching)\n");
             } else if (batchSize == 0) {
                 sb.append("No (Use \"-batchSize\" to setup batching)\n");
-            } else if (batchSize == -1) {
+            } else {
```
Comment (softwarejimenez) [inline srcJava/com/rti/perftest/harness/PerfTest.java]:
```
// < 0
```

File: srcCs/perftest_publisher.cs [hunk 6]
```diff
@@ -1387,10 +1387,16 @@ bool ParseConfig(string[] argv)
                     sb.Append(" Bytes (Use \"-batchSize 0\" to disable batching)\n");
                 } else if (batchSize == 0) {
                     sb.Append("No (Use \"-batchSize\" to setup batching)\n");
-                } else if (batchSize == -1) {
+                } else {
```
Comment (softwarejimenez) [inline srcCs/perftest_publisher.cs]:
```
 // < 0
```

File: srcJava/com/rti/perftest/ddsimpl/RTIDDSImpl.java (hunk 4 above)
Comment (javmorcas) [reply srcJava/com/rti/perftest/ddsimpl/RTIDDSImpl.java]: Done, actually it was "Batching cannot be used with Large Data." I fixed it.

File: srcJava/com/rti/perftest/harness/PerfTest.java (hunk 5 above)
Comment (javmorcas) [reply srcJava/com/rti/perftest/harness/PerfTest.java]: Done.

File: srcCs/perftest_publisher.cs (hunk 6 above)
Comment (javmorcas) [reply srcCs/perftest_publisher.cs]: Done.

**Pull request merged**